main.go

```diff
@@ -139,6 +151,28 @@ func configureTLS(certFile, keyFile string) (*tls.Config, error) {
 	}, nil
 }
 
+func configureLetsEncrypt(publicURL string) (tlsConfig *tls.Config, err error) {
+	parsedUrl, err := url.Parse(publicURL)
+	if err != nil {
+		return nil, err
+	}
+	certManager := autocert.Manager{
+		Prompt:     autocert.AcceptTOS,
+		HostPolicy: autocert.HostWhitelist(parsedUrl.Host),
+		Cache:      autocert.DirCache(".certs"),
+	}
+	tlsConfig = &tls.Config{
+		GetCertificate: certManager.GetCertificate,
+	}
+
+	// run HTTP Endpoint as redirect and challenge handler
+	go func() {
+		_ = http.ListenAndServe(":http", certManager.HTTPHandler(nil))
```

Comment:
This can clash with the link-sharing server if it's run to listen on the port 80 and it will error silently.

On the other hand, isn't there a better way of not running this uncontrolled goroutine so it can be stopped as the link-sharing server is?

Reply:
If you have a proposal, please make a PR. I did not come up in that short time crunsh

Reply:
Also this is only triggered, if you use SSL using lets encrypt, which should never listen on port 80

Reply:
I don't have a proposal except that I quickly thought, that's logging that error and make the user aware.

I'll create a ticket for it, something that I thought to do but I preferred to ask here.